### PR TITLE
[DOCS] Update 8.0.0-rc1 release notes

### PR DESCRIPTION
**Pull request by arteam**
Add the "bug fixes" section which for some reason was missed from the original PR.

### CODE REVIEW

File: docs/reference/release-notes/8.0.0-rc1.asciidoc [hunk 1]
```diff
@@ -93,5 +93,23 @@ Security::
...
 === Bug fixes
 
+Autoscaling::
+* Autoscaling use adjusted total memory {es-pull}80528[#80528] (issue: {es-issue}78750[#78750])
+
+Data streams::
+* Prohibit restoring a data stream alias with a conflicting write data stream {es-pull}81217[#81217] (issue: {es-issue}80976[#80976])
+
+ILM+SLM::
+* Less Verbose Serialization of Snapshot Failure in SLM Metadata {es-pull}80942[#80942] (issue: {es-issue}77466[#77466])
```
Comment (jrodewig) [inline docs/reference/release-notes/8.0.0-rc1.asciidoc:103]:
```suggestion
* Less verbose serialization of snapshot failure in SLM metadata {es-pull}80942[#80942] (issue: {es-issue}77466[#77466])
```

**Review by jrodewig — APPROVED** [review]
LGTM with a minor nit.

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------:|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/82495)|

**Comment by arteam** [comment]
Thank you!

**Pull request merged**